docs/users/configuration.md

```diff
@@ -57,3 +57,71 @@ triggered.rules = [
   DisableSyntax
 ]
 ```
+
+## Overriding the linting behavior
+
+By default, only lints raised with `LintSeverity.Error` will result in a
+non-zero exit code. To force Scalafix to return a non-zero exit code for other
+diagnostics, it is possible to promote non-error diagnostics to the error level
+without touching the rules(s) source code, thanks to `lint.error`.
+
+This key accepts as value either
+* a string, interpreted as a regular expression to match diagnostics using the
+  `<rule_name>[.<lint_id>]` pattern,
+* a list of strings, treated as several regular expressions like above,
+* an object, with `include` and `exclude` attributes, each treated as regular
+  expressions like above.
+
+As an example, to force all linter reports to trigger an error, except for unused
+suppressions, any of the 3 configuration files below can be used, as they are
+effectively equivalent.
+
+```scala
+// .scalafix.conf
+lint.error = "^((?!UnusedScalafixSuppression).)*$"
+```
+
+```scala
+// .scalafix.conf
+lint.error = [
+  "^((?!UnusedScalafixSuppression).)*$"
+]
+```
+
+```scala
+// .scalafix.conf
+lint.error = {
+  includes = ".*"
+  excludes = "UnusedScalafixSuppression"
+}
+```
+
+Similarly, diagnostics can be controlled through `lint.ignore`, `lint.info`,
+and `lint.warning`, which respectively suppress matching diagnostics, force
+them to be at info level, and force them to be at warning level.
+
+## Passing configuration as CLI arguments
+
+It is possible to pass configuration flags with scalar values as CLI arguments,
+overriding any potential value set in the configuration file, by prefixing the
+qualified name of each attribute with `--settings.`.
+
+For example, the configuration file below
+
+```scala
+// .scalafix.conf
+DisableSyntax {
+  noFinalize = true
+}
+lint.error.includes = ".*"
+lint.error.excludes = "UnusedScalafixSuppression"
+```
+
+is equivalent to the following CLI arguments
+
+```
+scalafix ... \
+  --settings.DisableSyntax.noFinalize=true \
+  --settings.lint.error.includes=.* \
```

Comment:
In a shell context, this needs to be wrapped in single quotes so that `*` is not processed.

```suggestion
  '--settings.lint.error.includes=.*' \
```

However, this syntax is not supported on sbt-scalafix, so I'd rather document the sbt-scalafix one, and let CLI users figure out by themselves (instead of being very verbose about it here and confusing sbt-scalafix users).

Reply:
It turns out that multiline syntax does not work on sbt, so I am taking back what I wrote above and will target CLI/shell syntax instead in another PR